packages/config/src/api/site_info.ts

```diff
@@ -178,34 +240,23 @@ export const getIntegrations = async function ({
   const url = accountId
     ? `${baseUrl}team/${accountId}/integrations/installations/meta/${siteId}`
     : `${baseUrl}site/${siteId}/integrations/safe`
+  const headers = new Headers({
+    'Netlify-Config-Mode': mode,
+    'User-Agent': `Netlify Config (mode:${mode}) / ${ROOT_PACKAGE_JSON.version}`,
+  })
+  if (sendBuildBotTokenToJigsaw && token) {
+    headers.set('Netlify-SDK-Build-Bot-Token', token)
+  }
 
   try {
-    const requestOptions = {} as RequestInit
-
-    // This is used to identify where the request is coming from
-    requestOptions.headers = {
-      'netlify-config-mode': mode,
+    const res = await fetch(url, { headers })
+    if (res.status !== 200) {
+      throw new Error(`Unexpected status code ${res.status} from fetching extensions`)
     }
-
-    if (sendBuildBotTokenToJigsaw && token) {
-      requestOptions.headers = {
-        ...requestOptions.headers,
-        'netlify-sdk-build-bot-token': token,
-      }
-    }
-
-    const response = await fetch(url, requestOptions)
-    if (!response.ok) {
```

Comment:
This check is fine, but we _know_ this should be a 200, so making this check more specific.